wrapper/src/vmaf.cpp

```diff
@@ -823,83 +827,117 @@ std::unique_ptr<LibsvmNusvrTrainTestModel> BootstrapVmafQualityRunner::_load_mod
-    double& scorePlusDelta = predictionMap[VmafPredictionReturnType::PLUS_DELTA];
-    double& scoreMinusDelta = predictionMap[VmafPredictionReturnType::MINUS_DELTA];
+    double& score = predictionStruct.vmafPrediction[VmafPredictionReturnType::SCORE];
+    double& baggingScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::BAGGING_SCORE];
+    double& ci95LowScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::CI95_LOW];
+    double& ci95HighScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::CI95_HIGH];
+    double& scorePlusDelta = predictionStruct.vmafPrediction[VmafPredictionReturnType::PLUS_DELTA];
+    double& scoreMinusDelta = predictionStruct.vmafPrediction[VmafPredictionReturnType::MINUS_DELTA];
 
     _transform_value(model, score);
     _transform_value(model, baggingScore);
     _transform_value(model, ci95LowScore);
     _transform_value(model, ci95HighScore);
     _transform_value(model, scorePlusDelta);
     _transform_value(model, scoreMinusDelta);
+
+    // transform bootstrap model scores
+    int num_models = predictionStruct.vmafMultiModelPrediction.size();
+
+    for (int model_i = 0; model_i < num_models; model_i++)
+    {
+        _transform_value(model, predictionStruct.vmafMultiModelPrediction.at(model_i));
+    }
 }
 
 void BootstrapVmafQualityRunner::_clip_score(LibsvmNusvrTrainTestModel& model,
-        std::map<VmafPredictionReturnType, double>& predictionMap) {
+        VmafPredictionStruct& predictionStruct) {
 
-    double& score = predictionMap[VmafPredictionReturnType::SCORE];
-    double& baggingScore = predictionMap[VmafPredictionReturnType::BAGGING_SCORE];
-    double& ci95LowScore = predictionMap[VmafPredictionReturnType::CI95_LOW];
-    double& ci95HighScore = predictionMap[VmafPredictionReturnType::CI95_HIGH];
-    double& scorePlusDelta = predictionMap[VmafPredictionReturnType::PLUS_DELTA];
-    double& scoreMinusDelta = predictionMap[VmafPredictionReturnType::MINUS_DELTA];
+    double& score = predictionStruct.vmafPrediction[VmafPredictionReturnType::SCORE];
+    double& baggingScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::BAGGING_SCORE];
+    double& ci95LowScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::CI95_LOW];
+    double& ci95HighScore = predictionStruct.vmafPrediction[VmafPredictionReturnType::CI95_HIGH];
+    double& scorePlusDelta = predictionStruct.vmafPrediction[VmafPredictionReturnType::PLUS_DELTA];
+    double& scoreMinusDelta = predictionStruct.vmafPrediction[VmafPredictionReturnType::MINUS_DELTA];
 
     _clip_value(model, score);
     _clip_value(model, baggingScore);
     _clip_value(model, ci95LowScore);
     _clip_value(model, ci95HighScore);
     _clip_value(model, scorePlusDelta);
     _clip_value(model, scoreMinusDelta);
+
+    // clip bootstrap model scores
+    int num_models = predictionStruct.vmafMultiModelPrediction.size();
+
+    for (int model_i; model_i < num_models; model_i++)
```

Comment:
Shouldn't `model_i` be initialized with value 0?

Reply:
good catch, thanks! I will be updating this together with some other fixes